pymongo/pool.py

```diff
@@ -1549,7 +1559,10 @@ def connect(self, handler: Optional[_MongoClientErrorHandler] = None) -> Connect
                 )
 
             if isinstance(error, (IOError, OSError, SSLError)):
-                _raise_connection_failure(self.address, error)
+                timeout_details = None
+                if handler:
+                    timeout_details = _get_timeout_details(handler.client)
+                _raise_connection_failure(self.address, error, details=timeout_details)
```

Comment:
Hmm is there a central place we could catch and re-raise these exceptions (rather than sprinkling format_timeout_details in various places)?

Reply:
Ideally we'd be able to access the `_MongoClientErrorHandler` whenever we catch a timeout exception. Are there existing limitations to our internal network interfaces that prevent us from passing an instance of the handler around more widely?

Reply:
Following up @ShaneHarvey

Reply:
I believe we do already wrap all operations with _MongoClientErrorHandler. Does that answer your question?

Reply:
`_MongoClientErrorHandler` is only wrapped around `_checkout` and `_run_operation` in MongoClient, are those the only two operations relevant for this issue? If so, we could maybe move the logic in `format_timeout_details` into the error handler itself to centralize the processing.

Reply:
Answered in Zoom but summarizing here:
Yes that's true but it's better we add the info at the point the exception is raised to ensure that event publishing and (soon) command logging also include the same info. If we catch and reraise the exception in _MongoClientErrorHandler then the events/logs will see a different error message than the user.